### DISCUSSION TITLE
fail to  build milvus v2.3.4 in centos 8 ARM v8

### DISCUSSION THREAD

**Discussion by Ghostbaby** [milvus-io/milvus#30801] [Q&A and General discussion]
gcc: 9.2.1
glibc: 2.28
kernel: Linux LS-209271-39728 4.18.0-408.el8.aarch64 #1 SMP Mon Jul 18 15:47:44 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
os: CentOS Stream release 8

errors:
`/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘float faiss::fvec_inner_product_neon(const float*, const float*, size_t)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:23:27: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
   23 |         float32x4x4_t a = vld1q_f32_x4(x + dim - d);
      |                           ^~~~~~~~~~~~
      |                           vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘float faiss::fvec_L2sqr_neon(const float*, const float*, size_t)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:90:27: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
   90 |         float32x4x4_t a = vld1q_f32_x4(x + dim - d);
      |                           ^~~~~~~~~~~~
      |                           vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘float faiss::fvec_L1_neon(const float*, const float*, size_t)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:169:27: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
  169 |         float32x4x4_t a = vld1q_f32_x4(x + dim - d);
      |                           ^~~~~~~~~~~~
      |                           vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘float faiss::fvec_Linf_neon(const float*, const float*, size_t)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:248:27: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
  248 |         float32x4x4_t a = vld1q_f32_x4(x + dim - d);
      |                           ^~~~~~~~~~~~
      |                           vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘float faiss::fvec_norm_L2sqr_neon(const float*, size_t)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:326:27: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
  326 |         float32x4x4_t a = vld1q_f32_x4(x + dim - d);
      |                           ^~~~~~~~~~~~
      |                           vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘void faiss::fvec_madd_neon(size_t, const float*, float, const float*, float*)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:400:19: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
  400 |         auto a_ = vld1q_f32_x4(a + len - n);
      |                   ^~~~~~~~~~~~
      |                   vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:411:9: error: ‘vst1q_f32_x4’ was not declared in this scope; did you mean ‘vst1q_f32_x3’?
  411 |         vst1q_f32_x4(c + len - n, c_);
      |         ^~~~~~~~~~~~
      |         vst1q_f32_x3
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc: In function ‘int faiss::fvec_madd_and_argmin_neon(size_t, const float*, float, const float*, float*)’:
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:488:19: error: ‘vld1q_f32_x4’ was not declared in this scope; did you mean ‘vld1q_f32_x2’?
  488 |         auto a_ = vld1q_f32_x4(a + len - n);
      |                   ^~~~~~~~~~~~
      |                   vld1q_f32_x2
/root/milvus/cmake_build/thirdparty/knowhere/knowhere-src/src/simd/distances_neon.cc:500:9: error: ‘vst1q_f32_x4’ was not declared in this scope; did you mean ‘vst1q_f32_x3’?
  500 |         vst1q_f32_x4(c + len - n, c_);
      |         ^~~~~~~~~~~~
      |         vst1q_f32_x3
make[3]: *** [thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere_utils.dir/build.make:104: thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere_utils.dir/src/simd/distances_neon.cc.o] Error 1
make[3]: Leaving directory '/root/milvus/cmake_build'
make[2]: *** [CMakeFiles/Makefile2:1332: thirdparty/knowhere/knowhere-build/CMakeFiles/knowhere_utils.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[  8%] Linking CXX static library ../../lib/libmilvus_proto.a`

**Comment by xiaofan-luan** [comment]
it seems that your arm cpu doesn't support some of the neon instructions.

Try to remove the definition of __ARM_NEON   see it helps

**Comment by xiaofan-luan** [comment]
@Ghostbaby 
could you give the community about some advice why this could happen
